lib/_stream_writable.js

```diff
@@ -497,67 +469,42 @@ function onwriteDrain(stream, state) {
 
 // If there's something in the buffer waiting, then process it
 function clearBuffer(stream, state) {
-  state.bufferProcessing = true;
-  var entry = state.bufferedRequest;
-
-  if (stream._writev && entry && entry.next) {
-    // Fast case, write everything using _writev()
-    var l = state.bufferedRequestCount;
-    var buffer = new Array(l);
-    var holder = state.corkedRequestsFree;
-    holder.entry = entry;
-
-    var count = 0;
-    var allBuffers = true;
-    while (entry) {
-      buffer[count] = entry;
-      if (!entry.isBuf)
-        allBuffers = false;
-      entry = entry.next;
-      count += 1;
-    }
-    buffer.allBuffers = allBuffers;
+  const buffered = state.buffered;
+  const bufferedLength = buffered.length;
 
-    doWrite(stream, state, true, state.length, buffer, '', holder.finish);
+  if (!bufferedLength || state.corked || state.bufferProcessing ||
+    state.writing) {
+    return;
+  }
 
-    // doWrite is almost always async, defer these to save a bit of time
-    // as the hot path ends with doWrite
+  state.bufferProcessing = true;
+  if (bufferedLength === 1) {
+    const { chunk, encoding, callback } = buffered[0];
```

Comment:
Destructuring like this is not optimized well enough in V8 yet, so we should probably avoid it here.

Reply:
Ditto for other places where this is added.